File: Servlets/Interface/db2/src/java/dbase2/musicdb/get_recommended_music_track.java
package dbase2.musicdb;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletResponse;

import java.io.IOException;
import java.io.PrintWriter;

import java.sql.Connection;
import java.sql.SQLException;
import java.sql.Statement;
import java.sql.DriverManager;
import java.sql.ResultSet;

public class get_recommended_music_track extends HttpServlet{
  
  public void doGet(HttpServletRequest request, HttpServletResponse response)
    throws ServletException, IOException{
    response.setContentType("text/html");
    PrintWriter writer = response.getWriter();
    writer.write("<html>\n");
    writer.write("<head><title>Get Recommended Music Tracks</title></head>\n");
    writer.write("<body>\n");
    writer.write("<center>\n");
    
    try{
      Class.forName("com.mysql.jdbc.Driver");
    }catch(ClassNotFoundException exc){
      exc.printStackTrace();
      printMsg("No JDBC driver found!", writer, request);
      return;
    }
    try{
      Connection connection = DriverManager.getConnection("jdbc:mysql://localhost/db2", "student", "student");
      Statement statement = connection.createStatement(); //referential integrity
      ResultSet result = null;
      
      
      /*Best rated music track*/
      result = statement.executeQuery("SELECT m.title, MAX(r.rating) FROM rating r, music m WHERE r.mid = m.mid");
       
      writer.write("<strong>Best rated music tracks</strong><br>\n");

    if(result != null){
      writer.write("<table border = 1>\n");
      writer.write("<tr>\n");
      writer.write("<td>Title</td>\n");
      writer.write("<td>Rating</td>\n");;
      writer.write("</tr>\n");
      
      while(result.next()){
        String title = result.getString("m.title");
        String rating = ((new Integer(result.getInt("r.rating"))).toString());

        writer.write("<tr>\n");
        writer.write("<td>"+title+"</td>\n");
        writer.write("<td>"+rating+"</td>\n");
        writer.write("</tr>\n");
      }
        writer.write("</table>\n");
        writer.write("<hr style=\"margin-left:20em; margin-right:20em\"><br>\n");
    }
    
      /*Best Rated Music Track Per Album*/
      result = statement.executeQuery("SELECT alb.name, m.title, r.rating FROM rating r, music m, album alb, attribute a WHERE r.mid = m.mid and m.aid = a.aid and a.albid = alb.albid GROUP BY alb.name HAVING r.rating > '0'");
       
      writer.write("<strong>Best rated music tracks per album</strong><br>\n");

    if(result != null){
      writer.write("<table border = 1>\n");
      writer.write("<tr>\n");
      writer.write("<td>Album</td>\n");
      writer.write("<td>Title</td>\n");
      writer.write("<td>Rating</td>\n");;
      writer.write("</tr>\n");
      
      while(result.next()){
        String album = result.getString("alb.name");
        String title = result.getString("m.title");
        String rating = ((new Integer(result.getInt("r.rating"))).toString());

        writer.write("<tr>\n");
        writer.write("<td>"+album+"</td>\n");
        writer.write("<td>"+title+"</td>\n");
        writer.write("<td>"+rating+"</td>\n");
        writer.write("</tr>\n");
      }
        writer.write("</table>\n");
        writer.write("<hr style=\"margin-left:20em; margin-right:20em\"><br>\n");
    }
    
      /*Best Rated Music Track Per Artist*/
      result = statement.executeQuery("SELECT art.name, m.title, r.rating FROM $table6 r, $table2 m, $table5 art, $table7 a WHERE r.mid = m.mid and m.aid = a.aid and a.artid = art.artid GROUP BY art.name HAVING r.rating > '0'");
       
      writer.write("<strong>Best Rated Music Track Per Artist</strong><br>\n");

    if(result != null){
      writer.write("<table border = 1>\n");
      writer.write("<tr>\n");
      writer.write("<td>Artist</td>\n");
      writer.write("<td>Title</td>\n");
      writer.write("<td>Rating</td>\n");;
      writer.write("</tr>\n");
      
      while(result.next()){
        String artist = result.getString("art.name");
        String title = result.getString("m.title");
        String rating = ((new Integer(result.getInt("r.rating"))).toString());

        writer.write("<tr>\n");
        writer.write("<td>"+artist+"</td>\n");
        writer.write("<td>"+title+"</td>\n");
        writer.write("<td>"+rating+"</td>\n");
        writer.write("</tr>\n");
      }
        writer.write("</table>\n");
        writer.write("<hr style=\"margin-left:20em; margin-right:20em\"><br>\n");
    }
    
      /*Best Rated Music Track Per Customer*/
      result = statement.executeQuery("SELECT c.name, m.title, r.rating FROM $table6 r, $table2 m, $table1 c WHERE r.mid = m.mid and r.cid = c.cid GROUP BY c.name HAVING r.rating > '0'");
       
      writer.write("<strong>Best rated music tracks per customer</strong><br>\n");

    if(result != null){
      writer.write("<table border = 1>\n");
      writer.write("<tr>\n");
      writer.write("<td>Customer</td>\n");
      writer.write("<td>Title</td>\n");
      writer.write("<td>Rating</td>\n");;
      writer.write("</tr>\n");
      
      while(result.next()){
        String customer = result.getString("c.name");
        String title = result.getString("m.title");
        String rating = ((new Integer(result.getInt("r.rating"))).toString());

        writer.write("<tr>\n");
        writer.write("<td>"+customer+"</td>\n");
        writer.write("<td>"+title+"</td>\n");
        writer.write("<td>"+rating+"</td>\n");
        writer.write("</tr>\n");
      }
        writer.write("</table>\n");
        writer.write("<hr style=\"margin-left:20em; margin-right:20em\"><br>\n");
    }

    printMsg("", writer, request);
    }catch(SQLException exc){
      exc.printStackTrace();
      printMsg("Database error!", writer, request);
    }
  }

  private void printMsg(String msg, PrintWriter writer, HttpServletRequest request){
    writer.write("<h1>" + msg + "</h1>\n");
    writer.write("</center>\n");
    writer.write("<hr style=\"margin-left:5em; margin-right:5em\">\n");
    writer.write("<center>\n");
    writer.write("<a href = \"index.html\">Back</a>\n");
    writer.write("</center>\n");
    writer.write("</body>\n");
    writer.write("</html>\n");
  }
}

File: PHP/Interface/get_recommended_music_track.php
<?
/* this script will handle the variables passed from the insert_customer.html file */

/* declare some relevant variables */

PRINT "<html>";
PRINT "<head><title>Get Recommended Music Tracks</title></head>";
PRINT "<body>";

include 'connection.php';

/* MySQL table created to store the data */
$table0 = "city";
$table1 = "customer";
$table2 = "music";
$table3 = "year";
$table4 = "album";
$table5 = "artist";
$table6 = "rating";
$table7 = "attribute";
$table8 = "genre";

PRINT "<center>";

  /*Best rated music track*/
  $query = "SELECT m.title, MAX(r.rating) FROM $table6 r, $table2 m WHERE r.mid = m.mid";
  $result = MYSQL_QUERY($query);

/* How many rows in the result? */
if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}
 PRINT "<strong>Best rated music tracks</strong><br>";
if ($number != 0)
{
 $i = 0;
 PRINT "<table border = 1>";
 PRINT "<tr>";
 PRINT "<td>Title</td>";
 PRINT "<td>Rating</td>";
 PRINT "</tr>";
 while ($i < $number){
    PRINT "<tr>";
    $title = mysql_result($result, $i, "m.title");
    PRINT "<td>$title</td>";
    $rating = mysql_result($result, $i, $i+1);
    PRINT "<td>$rating</td>";
    PRINT "</tr>";
    $i++;
 }
 PRINT "</table>";
 PRINT "<hr style=\"margin-left:20em; margin-right:20em\"><br>";
}
else
{
  PRINT "<strong>No Ratings in Database $table6</strong><br>";
  PRINT "<hr style=\"margin-left:20em; margin-right:20em\"><br>";
}
  /*Best Rated Music Track Per Album*/
  $query = "SELECT alb.name, m.title, r.rating FROM $table6 r, $table2 m, $table4 alb, $table7 a WHERE r.mid = m.mid and m.aid = a.aid and a.albid = alb.albid GROUP BY alb.name HAVING r.rating > '0'";
  $result = MYSQL_QUERY($query);

/* How many rows in the result? */
if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}
PRINT "<strong>Best rated music tracks per album</strong><br>";
if ($number != 0)
{
 $i = 0;
 PRINT "<table border = 1>";
 PRINT "<tr>";
 PRINT "<td>Album</td>";
 PRINT "<td>Title</td>";
 PRINT "<td>Rating</td>";
 PRINT "</tr>";
 while ($i < $number){
    PRINT "<tr>";
    $album = mysql_result($result, $i, "alb.name");
    PRINT "<td>$album</td>";
    $title = mysql_result($result, $i, "m.title");
    PRINT "<td>$title</td>";
    $rating = mysql_result($result, $i, "r.rating");
    PRINT "<td>$rating</td>";
    PRINT "</tr>";
    $i++;
 }
 PRINT "</table>";
 PRINT "<hr style=\"margin-left:20em; margin-right:20em\"><br>";
}
else
{
  PRINT "<strong>No Ratings in Database $table6</strong><br>";
  PRINT "<hr style=\"margin-left:20em; margin-right:20em\"><br>";
}

  /*Best Rated Music Track Per Artist*/
  $query = "SELECT art.name, m.title, r.rating FROM $table6 r, $table2 m, $table5 art, $table7 a WHERE r.mid = m.mid and m.aid = a.aid and a.artid = art.artid GROUP BY art.name HAVING r.rating > '0'";
  $result = MYSQL_QUERY($query);

/* How many rows in the result? */
if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}
PRINT "<strong>Best rated music tracks per artist</strong><br>";
if ($number != 0)
{
 $i = 0;
 PRINT "<table border = 1>";
 PRINT "<tr>";
 PRINT "<td>Artist</td>";
 PRINT "<td>Title</td>";
 PRINT "<td>Rating</td>";
 PRINT "</tr>";
 while ($i < $number){
    PRINT "<tr>";
    $artist = mysql_result($result, $i, "art.name");
    PRINT "<td>$artist</td>";
    $title = mysql_result($result, $i, "m.title");
    PRINT "<td>$title</td>";
    $rating = mysql_result($result, $i, "r.rating");
    PRINT "<td>$rating</td>";
    PRINT "</tr>";
    $i++;
 }
 PRINT "</table>";
 PRINT "<hr style=\"margin-left:20em; margin-right:20em\"><br>";
}
else
{
  PRINT "<strong>No Ratings in Database $table5</strong><br>";
  PRINT "<hr style=\"margin-left:20em; margin-right:20em\"><br>";
}
  /*Best Rated Music Track Per Customer*/
  $query = "SELECT c.name, m.title, r.rating FROM $table6 r, $table2 m, $table1 c WHERE r.mid = m.mid and r.cid = c.cid GROUP BY c.name HAVING r.rating > '0'";
  $result = MYSQL_QUERY($query);

/* How many rows in the result? */
if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}
PRINT "<strong>Best rated music tracks per customer</strong><br>";
if ($number != 0)
{
 $i = 0;
 PRINT "<table border = 1>";
 PRINT "<tr>";
 PRINT "<td>Customer</td>";
 PRINT "<td>Title</td>";
 PRINT "<td>Rating</td>";
 PRINT "</tr>";
 while ($i < $number){
    PRINT "<tr>";
    $customer = mysql_result($result, $i, "c.name");
    PRINT "<td>$customer</td>";
    $title = mysql_result($result, $i, "m.title");
    PRINT "<td>$title</td>";
    $rating = mysql_result($result, $i, "r.rating");
    PRINT "<td>$rating</td>";
    PRINT "</tr>";
    $i++;
 }
 PRINT "</table>";
 PRINT "<hr style=\"margin-left:20em; margin-right:20em\"><br>";
}
else
{
  PRINT "<strong>No Ratings in Database $table1</strong><br>";
  PRINT "<hr style=\"margin-left:20em; margin-right:20em\"><br>";
}

PRINT "</center>";
PRINT "<hr style=\"margin-left:5em; margin-right:5em\">";
PRINT "<center>";
PRINT "<a href = \"index.html\">Back</a>";
PRINT "</center>";
PRINT "</body>";
PRINT "</html>";

/* Close the database connection */
MYSQL_CLOSE();

?>
